gazebo/msgs/factory.proto

```diff
@@ -40,4 +40,7 @@ message Factory
   /// \brief Whether the server is allowed to rename the model in case of
   /// overlap with existing models.
   optional bool allow_renaming = 6 [default = true];
+
+  /// \brief New name of the entity
+  optional string new_name = 7;
```

Comment:
modifying protobuf messages usually breaks ABI

our ABI checking CI jobs have been broken for a while, though I think they were recently fixed; I'll try to run one for this branch

Reply:
[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gazebo-abichecker-any_to_any-ubuntu_auto-amd64&build=194)](https://build.osrfoundation.org/job/gazebo-abichecker-any_to_any-ubuntu_auto-amd64/194/) https://build.osrfoundation.org/job/gazebo-abichecker-any_to_any-ubuntu_auto-amd64/194/

Reply:
yeah, the ABI checker is not happy about this:

* https://build.osrfoundation.org/job/gazebo-abichecker-any_to_any-ubuntu_auto-amd64/194/API_5fABI_20report/

Reply:
Ouch!. I guess we'll have to modify the xml to set the new name of the model. That will complicate things, but I'll give it a shot.

Reply:
I removed the field in https://github.com/osrf/gazebo/pull/2734/commits/5a58db6d4730fcc64bac662165cd688291aeedae. The solution I came up with is to use tinyxml to set the new name of the model to be spawned.